From: Felix Krull <f_krull@gmx.de>
Date: Fri, 27 Apr 2018 08:06:23 -0700
Subject: deadsnakes-libdb-versions

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 45fd9ef..0485f0c 100644
--- a/setup.py
+++ b/setup.py
@@ -553,8 +553,8 @@ class PyBuildExt(build_ext):
         #
         # http://www.sleepycat.com/update/index.html
 
-        max_db_ver = (4, 4)
-        min_db_ver = (3, 2)
+        max_db_ver = (5, 3)
+        min_db_ver = (4, 8)
         db_setup_debug = False   # verbose debug prints from this script?
 
         def allow_db_ver(db_ver):
